### ISSUE TITLE
Skip the train_step based on batch statistics

### ISSUE THREAD

**Issue by pamparana34** [PyTorchLightning/pytorch-lightning#3323]
I have an issue where I want to skip a training step based on the input batch statistics. I am just wondering if it is possible to skip the `train_step` in a lightning model i.e. the training would just ignore this batch and carry on with the next one?

usually, I would make sure the batches are always generated in an acceptable manner but I have a hard to track bug which happens rarely and due to some deadline pressures, I would still like to continue the training by skipping this batch. Is there a way to do this with the lightning framework?

**Comment by Borda** [comment]
you can write an iff statement inside the training step and kind of do nothing if you decided to skip the training step...

**Comment by BostonLobster** [comment]
@Borda But training_step requires returning a loss dict, how should I deal with this required return value? I tried simply doing nothing will cause AttributeError in training_loop.py

**Comment by AsaphLightricks** [comment]
i'm having a similar issue, i want to be able to skip a batch if something goes wrong in my calculation. What should the train_step method return as to not break the overall training?

**Comment by deng-cy** [comment]
You can return `None` to skip a step.

**Comment by ydcjeff** [comment]
According to @deng-cy comments, you should be able to skip a batch now.

Closing it as it is resolved, feel free to reopen if needed

**Comment by amin-nejad** [comment]
I think this behaviour should be documented somewhere but I don't see it anywhere. Do others agree? Does it only apply to the `training_step` method or will returning `None` also skip the batch in related hooks e.g. `on_train_batch_start`
